py-polars/polars/series/series.py

```diff
@@ -483,6 +483,11 @@ def _comp(self, other: Any, op: ComparisonOperator) -> Series:
                 return ~self
 
         if isinstance(other, datetime) and self.dtype == Datetime:
+            time_zone = self.dtype.time_zone  # type: ignore[union-attr]
+            if str(other.tzinfo) != str(time_zone):
+                raise TypeError(
+                    f"Time zone of datetime '{other.tzinfo}' does not match Series timezone '{time_zone}'"
```

Comment:
I'd tweak the error message like so, otherwise at first glance it reads like the time zone _is_ the datetime, instead of being something that _belongs_ to it :)

Before: `f"Time zone of datetime '{other.tzinfo}' `
After: `f"Datetime time zone '{other.tzinfo}' `